packages/manager/modules/vps/src/migration/vps-migration.controller.js

```diff
@@ -0,0 +1,67 @@
+import assign from 'lodash/assign';
+import remove from 'lodash/remove';
+
+import { NEW_RANGE_LINK } from './vps-migration.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($state, $q, $translate, CucCloudMessage, VpsMigrationService) {
+    this.$state = $state;
+    this.$q = $q;
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.VpsMigrationService = VpsMigrationService;
+    this.NEW_RANGE_LINK = NEW_RANGE_LINK;
+  }
+
+  $onInit() {
+    this.selectedRows = null;
+    this.loadMessage();
+  }
+
+  getMigrationInfo(vps) {
+    return this.getMigrationDetails(vps.name, this.catalog)
+      .then((res) => {
+        const migrationPlan = this.VpsMigrationService.getMigrationPlan(
+          this.catalog,
+          res.model,
+        );
+        return assign({
```

Comment:
Why do we use `assign` here?

Reply:
This assigns the object to the main List object `vpsList`. And also used to derive name and description of both vps and migration plan